payments/voucher_manager.go

```diff
@@ -108,31 +114,36 @@ func (vm *VoucherManager) Receive(voucher Voucher) (*big.Int, error) {
 	if err != nil {
 		return &big.Int{}, err
 	}
-	if signer != status.channelPayer {
-		return &big.Int{}, fmt.Errorf("wrong signer: %+v, %+v", signer, status.channelPayer)
+	if signer != status.ChannelPayer {
+		return &big.Int{}, fmt.Errorf("wrong signer: %+v, %+v", signer, status.ChannelPayer)
 	}
-	status.currentBalance.Paid.Set(received)
-	remaining := big.NewInt(0).Sub(status.startingBalance, received)
-	status.currentBalance.Remaining.Set(remaining)
+	status.Paid.Set(received)
+	remaining := big.NewInt(0).Sub(status.StartingBalance, received)
+	status.Remaining.Set(remaining)
+
+	status.LargestVoucher = voucher
 
-	status.largestVoucher = voucher
+	err = vm.store.SetVoucherInfo(voucher.ChannelId, *status)
+	if err != nil {
+		return nil, err
+	}
 	return received, nil
 }
 
 // ChannelRegistered returns  whether a channel has been registered with the voucher manager or not
 func (vm *VoucherManager) ChannelRegistered(channelId types.Destination) bool {
-	_, ok := vm.channels.Load(channelId.String())
+	_, ok := vm.store.GetVoucherInfo(channelId)
 	return ok
 
 }
 
 // Balance returns the balance of the channel
-func (vm *VoucherManager) Balance(channelId types.Destination) (Balance, error) {
-	data, ok := vm.channels.Load(channelId.String())
+func (vm *VoucherManager) Balance(channelId types.Destination) (initial, paid, remaining *big.Int, err error) {
```

Comment:
I decided to remove the `Balance` struct and just embed `paid/remaining` in the `VoucherInfo`. As a result this function now returns named return values.  I think it cleans things up slightly but it might be a matter of preference.